hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -112,6 +112,68 @@ threadToken = assetClass (currencySymbol "hydra") (tokenName "token")
 policyId :: MintingPolicyHash
 (policyId, _) = first currencyMPSHash (unAssetClass threadToken)
 
+emptyTx :: ValidatedTx Era
+emptyTx =
+  ValidatedTx
+    { body = emptyTxBody
+    , wits =
+        TxWitness
+          { txwitsVKey = mempty
+          , txwitsBoot = mempty
+          , txscripts = mempty
+          , txdats = mempty
+          , txrdmrs = Redeemers Map.empty
+          }
+    , isValid = IsValid True -- REVIEW(SN): no idea of the semantics of this
```

Comment:
`isValid = True` means that the transaction is expected to pass phase-2 validations (i.e. Plutus scripts pass). This is because, while local nodes will generally speaking protect against submitted a non-valid transaction, it is still possible to bypass the local check and forcefully broadcast a non-valid transaction as an adversarial move to force nodes consuming CPU and memory resources.
The collateral comes as a protection mechanism; if a transaction fails phase-2 validations, the collateral is collected and the transaction is still added to the ledger but marked as `IsValid = False`. 

TL;DR; 

- isValid = True means that inputs are consumed and outputs produced.
- isValid = False means that collaterals are consumed and nothing is produced.

Reply:
@KtorZ Is there a reference I could add as documentation in the source code that explains what you just said? Or should I simply quote this?